src/Middleware/Buffer.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace React\Http\Middleware;
+
+use Psr\Http\Message\ServerRequestInterface;
+use React\Http\Response;
+use React\Promise\Stream;
+use React\Stream\ReadableStreamInterface;
+use RingCentral\Psr7\BufferStream;
+
+final class Buffer
```

Comment:
Not sure about the name, as `new Buffer()` doesn't really convey its responsibility? How about something like `RequestBufferer` (which doesn't sound perfect either).

Reply:
`RequestBodyBuffer` could work.

Reply:
+1 for `RequestBodyBuffer`.

Reply:
Done

Reply:
Sounds much better to me, so I won't block this :+1: 

To be this name still implies this class represents the buffer itself, while it is actually (only) responsible for the buffering process and storing the complete buffer back into the request instance that will be passed upstream.

I guess this class could use some documentation? :-)